v2-migration/src/main/java/software/amazon/awssdk/v2migration/S3PutObjectRequestToV2.java

```diff
@@ -0,0 +1,322 @@
+                                            + "Executors.newSingleThreadExecutor())).build()",
+                                            contentLength);
+
+            addTmImport("UploadRequest");
+            addAsyncRequestBodyImport();
+            addExecutorsImport();
+            return JavaTemplate.builder(v2Method).build()
+                               .apply(getCursor(), method.getCoordinates().replaceArguments(),
+                                      method.getArguments().get(0), inputStream);
+        }
+
+        private J.MethodInvocation addFileToPutObject(J.MethodInvocation method, Expression file) {
+            String v2Method = "#{any()}, RequestBody.fromFile(#{any()})";
+            return JavaTemplate.builder(v2Method).build()
+                                 .apply(getCursor(), method.getCoordinates().replaceArguments(),
+                                        method.getArguments().get(0), file);
+        }
+
+        private J.MethodInvocation addInputStreamToPutObject(J.MethodInvocation method, Expression inputStream) {
+            long contentLength = extractContentLengthIfSet(method);
+            String v2Method;
+
+            if (contentLength < 0) {
+                // CHECKSTYLE:OFF: Regexp
+                v2Method = "#{any()}, RequestBody.fromContentProvider(() -> #{any()}, \"binary/octet-stream\")";
+                // CHECKSTYLE:ON: Regexp
+            } else {
+                v2Method = String.format("#{any()}, RequestBody.fromInputStream(#{any()}, %d)", contentLength);
+            }
+            return JavaTemplate.builder(v2Method).build()
+                                 .apply(getCursor(), method.getCoordinates().replaceArguments(),
+                                        method.getArguments().get(0), inputStream);
+        }
+
+        private long extractContentLengthIfSet(J.MethodInvocation method) {
+            // TODO - check ObjectMetadata and return if set
+            // Need to implement ObjectMetadata transform
+
+            return -1;
+        }
+
+        private J.MethodInvocation transformRequesterPaysSetter(J.MethodInvocation method) {
+            Expression expression = method.getArguments().get(0);
+            if (expression instanceof J.Literal) {
+                J.Literal literal = (J.Literal) expression;
+                if (Boolean.TRUE.equals(literal.getValue())) {
+                    addS3Import("RequestPayer");
+                } else {
+                    // Removes method - there is no enum value for false
+                    return (J.MethodInvocation) method.getSelect();
+                }
+            }
+
+            return JavaTemplate.builder("RequestPayer.REQUESTER").build()
+                               .apply(getCursor(), method.getCoordinates().replaceArguments());
+        }
+
+        /** Field set during POJO instantiation, e.g.,
+         * PutObjectRequest request = new PutObjectRequest("bucket" "key", "redirectLocation").withFile(file);
```

Comment:
I was scratching my head cause we're comparing against the v2 FQCN. I assume it's because the types are migrated at this point?